src/transformers/integrations/executorch.py

```diff
@@ -306,18 +307,24 @@ def forward(self, input_ids: torch.Tensor, cache_position: torch.Tensor):
             The adapter matches the model's forward signature with that in `executorch/extension/llm/runner`,
             ensuring that the exported model can be executed in `ExecuTorch` out-of-the-box.
         """
-        _, seqlen = input_ids.shape
         position_ids = cache_position.unsqueeze(0)
         past_key_values = self.static_cache
-
-        outs = self.model(
-            input_ids=input_ids,
-            attention_mask=None,
-            position_ids=position_ids,
-            cache_position=cache_position,
-            past_key_values=past_key_values,
-            use_cache=True,
-        )
+        model_params = inspect.signature(self.model.forward).parameters
```

Comment:
Last thing can you make sure this works? Not fully sure inpect is already export compatible no?

Reply:
Yep and its also not very fast, but fine because we only export once!